pkg/draft/context.go

```diff
@@ -65,3 +71,17 @@ func newAppContext(s *Server, req *rpc.UpRequest, out io.Writer) (*AppContext, e
 		vals: vals,
 	}, nil
 }
+
+func getulid() string { return <-ulidc }
+
+// A channel which returns build ulids.
+var ulidc = make(chan string)
+
+func init() {
+	rnd := rand.New(rand.NewSource(time.Now().UTC().UnixNano()))
+	go func() {
+		for {
+			ulidc <- ulid.MustNew(ulid.Timestamp(time.Now().UTC()), rnd).String()
```

Comment:
Makes no difference to me and as far as I can reason, technically, there isn't much overhead. Because ulidc is an unbuffered channel, ULIDs will only be generated whenever getulid is invoked, then subsequent ulid will then queue up in the channel for the next invocation. The only overhead is the nominal stack size for the goroutine. In fact, i'd be curious to measure the performance to determine if using `<-ulidc` is more performant than `getulid`.